R/util-MakeSnapshotHelpers.R

```diff
@@ -410,17 +409,17 @@ MakeRptSiteKriDetails <- function(lResults, status_site, meta_workflow, meta_par
 #'
 #' @param lResults `list` the output from `Study_Assess()`
 #' @param config_workflow `data.frame` configuration workflow in lMeta argument of `Make_Snapshot()`
-#' @param gsm_analysis_date `string` Date of snapshot
+#' @param gsm_analysis_date `string` Date of snapshot. leave NULL if dealing with stacked results
 #'
 #' @export
 #'
 #' @keywords internal
-MakeRptKriBoundsDetails <- function(lResults, config_workflow, gsm_analysis_date) {
+MakeRptKriBoundsDetails <- function(lResults, config_workflow, gsm_analysis_date = NULL) {
   bounds <- MakeResultsBounds(lResults = lResults, dfConfigWorkflow = config_workflow)
   if (length(bounds) > 0) {
     bounds %>%
       mutate(
-        "snapshot_date" = gsm_analysis_date,
+        "snapshot_date" = {if(is.null(gsm_analysis_date)) "gsm_analysis_date" else gsm_analysis_date},
```

Comment:
Is there a reason to favor using `"gsm_analysis_date"` over `NA`/`NA_Date_`?

I think I'd prefer changing these to something like:

`"snapshot_date" = ifelse(is.null(gsm_analysis_date), NA, gsm_analysis_date)`

Reply:
I had made it this was as to assume that if gsm_analysis_date is missing from the function call then It assumes that column is in the data (lStackedSnapshots should already have gsm_analysis_date. and snapshot date would then be defined by that column not the character "gsm_analysis_date") otherwise if it's missing it would just return the character representation. This change was only implemented to convert a past lStackedSnapshots (calculated with Augment_Snapshot) without altering past dates